libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/TriggerConditions/OnCondition.cs

```diff
@@ -187,7 +187,7 @@ public void AddExternalCondition(string condition)
                 {
                     lock (this.extraConstraints)
                     {
-                        this.extraConstraints.Add(new ExpressionEngine().Parse(condition.TrimStart('=')));
+                        this.extraConstraints.Add(Expression.Parse(condition.TrimStart('=')));
                         this.fullConstraint = null; // reset to force it to be recalcaulated
```

Comment:
Not sure where this method is used--might it make sense to use StringExpression where it is used?